### ISSUE TITLE
ExceptEmojis attribute not working

### ISSUE THREAD

**Issue by oinochoe** [missive/emoji-mart#788]
```javascript
<Picker exceptEmojis={['japanese_goblin']}></Picker>
```

I wrote right that, but not working this attribute. please tell me why.
I use this version.

```
"emoji-mart": "^5.5.2",
"@emoji-mart/data": "^1.1.2",
"@emoji-mart/react": "^1.1.1",
```

**Comment by j4hangir** [comment]
I looked at the code, it seems partially implemented but if you search for the emoji it will still show up.

**Comment by oinochoe** [comment]
@j4hangir yeah, i can see the emoji disappear in the list, but it still show whenever searching.
I wonder if it was your intention.

**Comment by oinochoe** [comment]
> I looked at the code, it seems partially implemented but if you search for the emoji it will still show up.

@j4hangir yeah, i can see the emoji disappear in the list, but it still show whenever searching.
I wonder if it was your intention.

**Comment by j4hangir** [comment]
It should be excluded everywhere. I'd say it's overall poorly implemented, for one there isn't much point of passing `emoji.id` where one can simply pass the emoji itself. 

I would do a PR when I get back to it.

**Comment by EtienneLem** [comment]
This is most likely because `init` is called beforehand with `data`, is that possible? including the `exceptEmojis` in the `init` call would resolve this issue (for now):

```js
init({ data, exceptEmojis: ['japanese_goblin'] })
```

This is caused by the fact that we try to limit having to process all the data all the time; we don’t recreated the search index on every mount. That being said, it wouldn’t be too hard to detect when the search should be recreated, we already have some checks in place.

> I'd say it's overall poorly implemented, for one there isn't much point of passing `emoji.id` where one can simply pass the emoji itself.

I don’t see how it would be better from a performance perspective. For most people, the emoji object itself isn’t stored / known before initializing a picker.

**Comment by j4hangir** [comment]
@EtienneLem in my case it's passed when doing init(), and it's excluded, but when searching still pops up.

Excluding emojis as they're written is way more intuitive, I'd pass a list like ['❄️', '🍾'], for instance them being the given reactions to narrow down the choices, yet with the current implementation I'd need to find out what their IDs are as an extra unnecessary step.

Even the name is chosen badly tbh, it should simply be **exclude** imo.

**Comment by oinochoe** [comment]
> Even the name is chosen badly tbh, it should simply be exclude imo.
I agree with this. 
@EtienneLem I solved the problem. but I think that "Readme" needs to be updated. Thanks 😄

**Comment by denisveli** [comment]
@oinochoe How did you solved the problem?

**Comment by oinochoe** [comment]
> @oinochoe How did you solved the problem?

@denisveli
https://github.com/missive/emoji-mart/issues/788#issuecomment-1412991040
https://github.com/missive/emoji-mart/issues/788#issuecomment-1416786959

I solved this issue via comments above.

**Comment by githorse** [comment]
I was tripped up by `exceptEmojis` as well because I passed in ids with the colons around them, like this:

```
exceptEmojis={[':+1:']}
```

... when that should have been
```
exceptEmojis={['+1']}
```

It would be nice to document that in the README. Right now it says "List of emoji IDs that will be excluded from the picker" -- it's very unclear `+1` is the ID, not `:+1:`. (Maybe the README API spec could include an "Example" column?)

Or, even better, just accept both versions, as well as the displayed emoji itself (as discussed above).
